lms/services/grouping/service.py

```diff
@@ -150,11 +174,46 @@ def get_course_groupings_for_user(
 
         return query.all()
 
+    def get_sections(self, lti_user, course, grading_student_id=None):
+        if not self.plugin.sections_type:
+            return None
+
+        if lti_user.is_learner:
+            groupings = self.plugin.get_sections_for_learner(self, course)
+
+        elif grading_student_id:
+            groupings = self.plugin.get_sections_for_grading(
+                self, course, grading_student_id
+            )
+
+        else:
+            groupings = self.plugin.get_sections_for_instructor(self, course)
+
+        return self._to_groupings(groupings, course, self.plugin.sections_type)
+
+    def get_groups(self, lti_user, course, group_set_id, grading_student_id=None):
+        if not self.plugin.group_type:
+            return None
+
+        if lti_user.is_learner:
+            groupings = self.plugin.get_groups_for_learner(self, course, group_set_id)
+
+        elif grading_student_id:
+            groupings = self.plugin.get_groups_for_grading(
+                self, course, group_set_id, grading_student_id
+            )
+
+        else:
+            groupings = self.plugin.get_groups_for_instructor(
+                self, course, group_set_id
+            )
+
+        return self._to_groupings(groupings, course, self.plugin.group_type)
```

Comment:
These two functions are annoyingly similar, but until we can do something about `group_set_id` this split looks here to stay.